kodiak.go

```diff
@@ -83,16 +92,18 @@ func (k *KodiakLPPriceProvider) Initialize(ctx context.Context, client *ethclien
 		return err
 	}
 
-	// Try to call the version() function... if error then not an Island V3 contract
-	_, err = k.contract.Version(opts)
-	if err != nil {
-		// If not a V3 Island, initialize it as a Kodiak V2 Pool
+	contractType := checkContractType(ctx, client, k.address)
+	if contractType == ContractTypeV2Pool {
```

Comment:
`switch` is a better approach here. Also, you can make the code simpler if you get the contract hash and here you use a switch with the contract hashes, so no need to have a type enum at all.